HPCC-16013 Add a config option to enable/disable Views Security Feature

RussWhitehead opened the pull request:
Added a new boolean "checkViewPermissions" flag to configuration. When set,
this signals ESP to enable the view scope feature that verifies users have permission to
access file/column pairs referenced in a view-enabled query (ESP modifications via
a different Jira/PR)

Signed-off-by: Russ Whitehead william.whitehead@lexisnexis.com

RussWhitehead commented:
@garonsky Please review

HPCCSmoketest commented:
Automated Smoketest 
Sha: 584a2773d11cd0927c2cb5bcb9d0f7d5d02e36a0
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:82 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

garonsky commented on initfiles/componentfiles/configxml/esp.xsd.in:
```diff
@@ -314,6 +314,15 @@
                                 </xs:appinfo>
                             </xs:annotation>
                         </xs:attribute>
+
```
Extra space

garonsky commented on initfiles/componentfiles/configxml/esp.xsd.in:
```diff
@@ -314,6 +314,15 @@
+                            </xs:appinfo>
+                          </xs:annotation>
+                        </xs:attribute>
+
```
extra space

garonsky commented:
Extra lines but otherwise looks good.

RussWhitehead committed:
HPCC-16013 Add a config option to enable/disable Views Security Feature

Added a new boolean "checkViewPermissions" flag to configuration. When set,
this enables the view scope feature that verifies users have permission to
access file/column pairs referenced in a view-enabled query

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

RussWhitehead commented:
@garonsky Removed blank lines
@ghalliday Please merge

HPCCSmoketest commented:
Automated Smoketest 
Sha: e6a8f35770d017cce696e0d9f3b513db5fc2a9f2
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:82 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

ghalliday commented:
@RussWhitehead which PR contains the check?

Jira: https://track.hpccsystems.com/browse/HPCC-16013

RussWhitehead commented:
@ghalliday The check for this flag has not been added yet, @sukhong will add it next week
